utils/virtual_machines.py

```diff
@@ -52,11 +59,23 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
     data = provider_crud.get_yaml_data()
 
     deploy_args.update(vm_name=vm_name)
+    if template_name is None:
+        try:
+            deploy_args.update(template=data['small_template'])
+        except KeyError:
+            raise ValueError('Provider {} does not have small_template defined in yaml').format(
+                provider_key)
+    else:
+        deploy_args.update(template=template_name)
     if isinstance(mgmt, RHEVMSystem):
         if 'default_cluster' not in deploy_args:
             deploy_args.update(cluster=data['default_cluster'])
     elif isinstance(mgmt, VMWareSystem):
-        if "allowed_datastores" not in deploy_args and "allowed_datastores" in data:
+        if 'resource_pool' in data["cap_and_util"] and purpose == 'capandu':
```

Comment:
Having a look at this I'm not keen on the ```purpose == capandu``` being responsible for the resourcepool being set. Can we not just set the ```resourcepool``` deployarg and leave it at that? I mean the call from the function above

```deploy_template(provider, vm_name, template, purpose)```

becomes

```deploy_template(provider, vm_name, template, resourcepool=data['cap_and_util']['resource_pool'], allowed_datastores=data['cap_and_util']['allowed_datastores'])```
Then we don't need to make any changes in this region.